File: Leetcode/Easy/628. Maximum Product of Three Numbers.py
# Author: DanPlus6 // David Fu
# Problem Name: 628. Maximum Product of Three Numbers
# Problem Code: maximum-product-of-three-numbers
# Problem URL: https://leetcode.com/problems/maximum-product-of-three-numbers/description

class Solution:
    def maximumProduct(self, nums: list[int]) -> int:
        return max((nums:=sorted(nums))[-1]*nums[-2]*nums[-3], nums[0]*nums[1]*nums[-1])

File: Leetcode/Easy/628. Maximum Product of Three Numbers.rb
# Author: DanPlus6 // David Fu
# Problem Name: 3536. Maximum Product of Two Digits
# Problem Code: maximum-product-of-two-digits
# Problem URL:  https://leetcode.com/problems/maximum-product-of-two-digits/

# @param {Integer[]} nums
# @return {Integer}
def maximum_product(nums)
    [(nums=nums.sort)[-1]*nums[-2]*nums[-3], nums[0]*nums[1]*nums[-1]].max
end
